### PR TITLE
Handle unexpected property type suggester errors in the frontend

### PR DESCRIPTION
**Pull request by benwerner01**
## 🌟 What is the purpose of this PR?

This PR adds basic error handling to the HASH FE for the property type suggester on the create entity type page. If any error is thrown, a generic error message is thrown with the ability to retry generating the property types.

## 🔗 Related links

- [Add error handling to property type suggester](https://app.asana.com/0/1204260760205594/1204461641771566/f) _(internal)_
- [Add "New" and "Existing" labels to property type suggester](https://app.asana.com/0/1204260760205594/1204461641771570/f) _(internal)_

## 🔍 What does this change?

- modifies the GQL resolver for calling AI agents to throw errors as GQL errors.

## 🚀 Has this modified a publishable library?

This PR:

- [ ] modifies an **npm**-publishable library and **I have added a changeset file(s)**
- [ ] modifies a **Cargo**-publishable library and **I have amended the version**
- [ ] modifies a **Cargo**-publishable library, but **it is not yet ready to publish**
- [ ] modifies a **block** that will need publishing via GitHub action once merged
- [x] does not modify any publishable blocks or libraries, or modifications do not need publishing
- [ ] I am unsure / need advice

## 📜 Does this require a change to the docs?

Not that I'm aware of.

## ⚠️ Known issues

- need to fix python linting error which wasn't caught previously

## 🛡 What tests cover this?

None

## ❓ How to test this?

1. Checkout the branch
2. Add an `Exception` to `/apps/hash-agents/app/agents/generate-entity-type-property-types/__main__.py`, for example by adding `raise Exception("Test Error")` to line 21
3. Use the create entity type page to try and generate property types
4. Observe the error state, and click retry. The generation should be retried, and will fail again due to the previous exception

## 📹 Demo

(You will want to skip through the middle section where I'm restarting the docker environment so that the python changes take effect)

https://user-images.githubusercontent.com/42802102/234626107-c05e09d7-1443-4b6d-856c-dae1b377407b.mp4

At the request of @nonparibus, I've also added "Existing" and "New" tags to the property types to indicate whether they already existed or not:
image

### CODE REVIEW

File: apps/hash-frontend/src/pages/new/types/entity-type/select-generated-property-types.tsx [hunk 1]
```diff
@@ -228,31 +228,48 @@ export const SelectGeneratedPropertyTypes: FunctionComponent<
...
+                        {description?.slice(1)}
+                        <Box
+                          component="span"
+                          sx={{
+                            marginLeft: 1,
+                            color: ({ palette }) => palette.gray[50],
+                          }}
+                        >
+                          <strong>
+                            <i>{isExistingType ? "Existing" : "New"}</i>
```
Comment (nonparibus) [inline apps/hash-frontend/src/pages/new/types/entity-type/select-generated-property-types.tsx:259]: @benwerner01 If something is "Existing" we don't need to write "Existing". Instead, can we include its path (e.g. `@hash/types/entity-type/Person`)?

Comment (benwerner01) [reply apps/hash-frontend/src/pages/new/types/entity-type/select-generated-property-types.tsx:259]: Sure, do you mean like this?

[the remark this answers is not in the document]
Comment (benwerner01) [reply apps/hash-frontend/src/pages/new/types/entity-type/select-generated-property-types.tsx:259]: I'm having trouble finding it, can you double check the link is correct?

[the remark this answers is not in the document]
Comment (benwerner01) [reply apps/hash-frontend/src/pages/new/types/entity-type/select-generated-property-types.tsx:259]: Have resolved, it now looks like this:
image

Comment (nonparibus) [reply apps/hash-frontend/src/pages/new/types/entity-type/select-generated-property-types.tsx]: Yes, either this or the existing chip. Thanks

[the remark this answers is not in the document]
Comment (nonparibus) [reply apps/hash-frontend/src/pages/new/types/entity-type/select-generated-property-types.tsx]: Great, thanks!

**Pull request merged**